mapie/classification.py

```diff
@@ -658,95 +859,147 @@ def predict(
         include_last_label = self._check_include_last_label(include_last_label)
         alpha_ = check_alpha(alpha)
         check_is_fitted(self, self.fit_attributes)
         if self.image_input:
             check_input_is_image(X)
         X = check_array(
-            X, force_all_finite=False, ensure_2d=self.image_input,
-            allow_nd=self.image_input, dtype=["float64", "object"]
+            X,
+            force_all_finite=False,
+            ensure_2d=self.image_input,
+            allow_nd=self.image_input,
+            dtype=["float64", "object"]
         )
+
+        # Estimate probabilities from estimator(s)
         y_pred = self.single_estimator_.predict(X)
-        y_pred_proba = self.single_estimator_.predict_proba(X)
-        y_pred_proba = self._check_proba_normalized(y_pred_proba)
+        if cv == "prefit":
+            y_pred_proba = self.single_estimator_.predict_proba(X)
+            y_pred_proba = self._check_proba_normalized(y_pred_proba)
+        else:
+            y_pred_proba_k = Parallel(
+                n_jobs=self.n_jobs, verbose=self.verbose
+            )(
+                delayed(self._predict_oof_model)(est, X)
+                for est in self.estimators_
+            )
+            if agg_scores == "crossval":
+                y_pred_proba = np.stack(
+                    [y_pred_proba_k[k] for k in self.k_], axis=0
+                )
+                y_pred_proba = self._check_proba_normalized(
+                    y_pred_proba, axis=2
+                )
+            elif agg_scores == "mean":
+                y_pred_proba = np.mean(y_pred_proba_k, axis=0)
+                y_pred_proba = self._check_proba_normalized(
+                    y_pred_proba, axis=1
+                )
+            else:
+                raise ValueError("Invalid 'agg_scores' argument.")
+
+        # Estimate prediction sets
         n = self.n_samples_val_
         if alpha_ is None:
             return np.array(y_pred)
-        else:
 
+        else:
             # Choice of the quantile
             check_alpha_and_n_samples(alpha_, n)
             if self.method == "naive":
                 self.quantiles_ = 1 - alpha_
             else:
-                self.quantiles_ = np.stack([
-                    np.quantile(
-                        self.conformity_scores_,
-                        ((n + 1) * (1 - _alpha)) / n,
-                        interpolation="higher"
-                    ) for _alpha in alpha_
-                ])
+                if (cv == "prefit") or (agg_scores in ["mean"]):
+                    self.quantiles_ = np.stack([
+                        np.quantile(
+                            self.conformity_scores_,
+                            ((n + 1) * (1 - _alpha)) / n,
+                            interpolation="higher"
+                        ) for _alpha in alpha_
+                    ])
+                else:
+                    self.quantiles_ = (n + 1) * (1 - alpha_)
```

Comment:
This is not a quantile right ? Is it normal ?

Reply:
Yes, so we can compare the *total* number of times the training scores are higher than the test scores.